renderer.js

```diff
@@ -6,6 +6,101 @@ the purpose of this file is to hold these objects and to make sure they are corr
 import * as THREE from 'three';
 import {EffectComposer} from 'three/examples/jsm/postprocessing/EffectComposer.js';
 import {minFov} from './constants.js';
+import {init, vec3, vec4, mat4} from 'glmw';
+
+// https://stackoverflow.com/a/62544968/3596736
+// Adding 'support' for instanceof Proxy:
+(() => {
+  var proxyInstances = new WeakSet();
+
+  // Optionally save the original in global scope:
+  window.OriginalProxy = Proxy;
+
+  window.Proxy = new Proxy(Proxy, {
+    construct(target, args) {
+      var newProxy = new window.OriginalProxy(...args);
+      proxyInstances.add(newProxy);
+      return newProxy;
+    },
+    get(obj, prop) {
+      if (prop === Symbol.hasInstance) {
+        return instance => {
+          return proxyInstances.has(instance);
+        };
+      }
+      return Reflect.get(...arguments);
+    },
+  });
+})();
+
+init().then(() => {
+  THREE.Matrix4.prototype.multiplyMatrices = (function() {
+    const mat4a = mat4.create();
+    const mat4b = mat4.create();
+    const mat4c = mat4.create();
+    const mat4aView = mat4.view(mat4a);
+    const mat4bView = mat4.view(mat4b);
+    const mat4cView = mat4.view(mat4c);
+
+    return function multiplyMatrices() {
+      if (Array.isArray(this.elements)) {
+        // NOTE: Replace the ordinary array with Float32Array to support the direct overall assignment by calling the `set` method, instead of assigning the 16 elements one by one.
+        this.elements = new Float32Array(this.elements);
+      } else if (this.elements instanceof Proxy) {
+        // NOTE: `localPlayer.avatar.modelBones.Root.matrixWorld.elements` will be replaced with `Proxy` after loading is complete, causing an error in calling the `set` method.
+        // I guess it is React's binding mechanism that generates the `Proxy`. After removing the `Proxy` in the following way, the main game part seems to still work normally, but I am not sure what effect it will have on the UI logic.
+        this.elements = new Float32Array(Object.values(this.elements));
+      }
+
+      mat4aView.set(arguments[0].elements);
```

Comment:
There should be no need to copy elements.

Reply:
Now no any data copying between js and wasm.